lte/gateway/python/magma/pipelined/app/he.py

```diff
@@ -0,0 +1,285 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from collections import namedtuple
+from typing import List
+
+from ryu.lib.packet import ether_types
+from ryu.lib.packet.in_proto import IPPROTO_TCP
+
+from .base import MagmaController, ControllerType
+from magma.pipelined.openflow import flows
+from magma.pipelined.openflow.magma_match import MagmaMatch
+from magma.pipelined.openflow.registers import load_direction, Direction, \
+    load_passthrough, set_proxy_tag, set_in_port, load_imsi, \
+    PROXY_TAG_TO_PROXY, set_tun_id
+
+from magma.pipelined.bridge_util import BridgeTools, DatapathLookupError
+from magma.pipelined.policy_converters import get_ue_ip_match_args, \
+    get_eth_type
+
+import logging
+
+
+PROXY_PORT_NAME = 'proxy_port'
+HTTP_PORT = 80
+PROXY_TABLE = 'proxy'
+
+
+class HeaderEnrichmentController(MagmaController):
+    """
+    A controller that tags related HTTP proxy flows.
+
+        1. From UE to Proxy
+        2. From Proxy to UE
+        3. From Proxy to Upstream server
+        4. From upstream server to Proxy
+
+    This controller is also responsible for setting direction for traffic
+    egressing proxy_port.
+    deatils are in: docs/readmes/proposals/p006_header_enrichment.md
+
+    """
+
+    APP_NAME = "proxy"
+    APP_TYPE = ControllerType.PHYSICAL
+
+    UplinkConfig = namedtuple(
+        'heConfig',
+        ['he_proxy_port',
+         'he_enabled',
+         'uplink_port',
+         'gtp_port'],
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(HeaderEnrichmentController, self).__init__(*args, **kwargs)
+
```

Comment:
Can you define public vars first, _ private vars last?also remove all the extra whitespace in between the vars